### PR TITLE
Add LASER plugin for whitelisting and blacklisting functions

### PR DESCRIPTION
**Pull request by b-mueller**
Here is a PR from the airport. It adds a plugin that allows whitelisting or blacklisting of functions. Also adds two new CLI arguments `--func-blacklist` and `--func-whilelist`. E.g.:

```
$myth a test.sol --func-blacklist "guess(uint8),refund(address)"
```

This addresses #1019 and is a prerequisite for adding better formal verification support.

**Known limitation:**

"fallback" is not affected by the lists because Mythril currently does not differentiate between the function prologue and the actual fallback function (i.e. if the user filters "fallback" then Mythril executes nothing at all).

### CODE REVIEW

File: mythril/interfaces/cli.py [hunk 1]
```diff
@@ -556,6 +569,17 @@ def execute_command(
             print("Disassembly: \n" + disassembler.contracts[0].get_creation_easm())
 
     elif args.command in ANALYZE_LIST:
+
+        if args.func_whitelist:
+            _whitelist = args.func_whitelist.split(",")
```
Comment (norhh) [inline mythril/interfaces/cli.py]: There can be multiple args, so we can't split by comma. `batchTransfer(uint a, uint b)` function will be split

**Comment by norhh** [comment]
I don't think this can work.
As code will be like 
```
func1_space
func2_space
...
```
if we skip state at one function(say func2) then we will skip all the below functions. I think we should constrain calldata for this.

**Comment by gkevinyen5418** [comment]
Constraining the calldata should work. I am working on a feature similar to this recently. Basically, I let user specify a list of functions of interest, then only verifies transaction sequences which calls one of them at least once.

I also generate callgraph from solidity AST, so that transaction sequences indirectly calling them are also verified.

This should be useful for users changing parts of their contracts and do not want to reverify the contract entirely.

**Comment by b-mueller** [comment]
Yeah you're right, constraining the calldata is a better way to do this.

@gkevinyen5418 would you like to contribute this feature?

**Comment by gkevinyen5418** [comment]
I would love to.

However, the feature I proposed requires source-code-level callgraphs (To resolve dependencies among functions).
I tried to use [surya](https://github.com/ConsenSys/surya) for this purpose, but it is unable to handle overloading and inheritance correctly.

As an alternative, I wrote my own AST visitor to do so. I will try to create a pull request to [solidity](https://github.com/ethereum/solidity) recently and see what happens.

**Pull request closed without merging**